### PR TITLE
chore(single-instance): put deep link integration behined a feature

### LINKED ISSUE #1017: [single-instance] doesn't shutdown immediately on Windows
When using single instance plugin on Windows, the second instance doesn't shutdown immediately

Windows implementation uses `app.exit` while the Linux one uses `std::process::exit`, `app.exit` only sends a shutdown signal, and the app will still be running through all the following setup functions, e.g. create the window, create the system tray...

This was changed to this in 4d903f3533f701441e06e83c4b666f1eef035d34 for cleaning up system tray, but currently it doesn't even trigger exit event for some reason (even if it actually triggers it, it's still problematic to have flashing windows)

In my opinion, we should just use `std::process::exit` here (maybe run with `cleanup_before_exit` to clean up system tray) (simple fix but may not be perfect) or we'll need a way to shutdown the app cleanly in setup step from Tauri

https://github.com/tauri-apps/plugins-workspace/blob/99bea2559c2c0648c2519c50a18cd124dacef57b/plugins/single-instance/src/platform_impl/windows.rs#L65

https://github.com/tauri-apps/plugins-workspace/blob/99bea2559c2c0648c2519c50a18cd124dacef57b/plugins/single-instance/src/platform_impl/linux.rs#L73

### PR DESCRIPTION
**Pull request by Legend-Master**
Also added some docs on the order of registering these 2 plugins

### CODE REVIEW

File: plugins/deep-link/README.md [hunk 1]
```diff
@@ -145,7 +145,7 @@ await onOpenUrl((urls) => {
 })
 ```
 
-Note that the Plugin will only emit events on macOS, iOS and Android. On Windows and Linux the OS will spawn a new instance of your app with the URL as a CLI argument. If you want your app to behave on Windows & Linux similar to the other platforms you can use the [single-instance](../single-instance/) plugin.
+Note that the Plugin will only emit events on macOS, iOS and Android. On Windows and Linux the OS will spawn a new instance of your app with the URL as a CLI argument. If you want your app to behave on Windows & Linux similar to the other platforms you can use the [single-instance](../single-instance/) plugin with deep-link feature.
```
Comment (FabianLars) [inline plugins/deep-link/README.md]:
```suggestion
Note that the Plugin will only emit events on macOS, iOS and Android. On Windows and Linux the OS will spawn a new instance of your app with the URL as a CLI argument. If you want your app to behave on Windows & Linux similar to the other platforms you can use the [single-instance](../single-instance/) plugin with the `deep-link` feature enabled.
```

File: plugins/single-instance/README.md [hunk 2]
```diff
@@ -49,6 +49,8 @@ fn main() {
 }
 ```
 
+Note that currently, plugins run in the order they were added in to the builder, you need to register this plugin first if you don't want the other plugins to run on the new instance
```
Comment (FabianLars) [inline plugins/single-instance/README.md]:
```suggestion
Note that currently, plugins run in the order they were added in to the builder, you need to register this plugin first if you don't want the other plugins to run on the temporary instances.
```

really not sure about this one, "new" sounds wrong to me. Temporary isn't much better though, maybe secondary? (secondary is what i use in discussions on discord at least but idk if it's good)

Comment (Legend-Master) [reply plugins/single-instance/README.md]: what about `non-main instances`? they all sound weird if there's no context explaining how this thing actually works to be honest

Comment (FabianLars) [reply plugins/single-instance/README.md]: We can also leave out the explanation (for now?) and just do
```
Note that currently, plugins run in the order they were added in to the builder, so make sure that this plugin is registered first.
```

**Review by FabianLars — APPROVED** [review]

File: plugins/single-instance/src/lib.rs [hunk 3]
```diff
@@ -35,7 +34,8 @@ pub fn init<R: Runtime, F: FnMut(&AppHandle<R>, Vec<String>, String) + Send + Sy
     mut f: F,
 ) -> TauriPlugin<R> {
     platform_impl::init(Box::new(move |app, args, cwd| {
-        if let Some(deep_link) = app.try_state::<DeepLink<R>>() {
+        #[cfg(feature = "deep-link")]
+        if let Some(deep_link) = app.try_state::<tauri_plugin_deep_link::DeepLink<R>>() {
```
Comment (lucasfernog) [inline plugins/single-instance/src/lib.rs:38]: actually for this to work I believe the deep link plugin must be registered first - so the DeepLink state has been initialized

Comment (lucasfernog) [reply plugins/single-instance/src/lib.rs:38]: though maybe that's not the case?? i remember it worked on linux idk

Comment (Legend-Master) [reply plugins/single-instance/src/lib.rs:38]: This is only called when the second instance notified the first one, and it's run on the first instance which should have initialized all the plugins at this point, not sure if it's going to cause problem if we spawned the second instances too fast though

**Pull request merged**